src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -557,6 +557,132 @@ public int Length
+        /// </summary>
+        public struct ChunkEnumerable
+        {
+            /// <summary>
+            /// Implements the IEnumerable pattern.  
+            /// </summary>
+            public ChunkEnumerator GetEnumerator() => new ChunkEnumerator(this._stringBuilder);
+
+            #region private
+            internal ChunkEnumerable(StringBuilder stringBuilder)
+            {
+                Debug.Assert(stringBuilder != null);    // Because it is only called with at 'this' pointer.    
+                _stringBuilder = stringBuilder;
+            }
+            private StringBuilder _stringBuilder;   // We insure this is never null. 
+            #endregion
+        }
+
+        /// <summary>
+        /// ChunkEnumerator supports the IEnumerator pattern so foreach works (see EnumerateChunks)  
+        /// It needs to be public (so the compiler can use it when building a foreach statement) 
+        /// but users typically don't use it explicitly (which is why it is a nested type).  
+        /// </summary>
+        public struct ChunkEnumerator
+        {
+            /// <summary>
+            /// Implements the IEnumerator pattern.
+            /// </summary>
+            public bool MoveNext()
+            {
+                if (_currentChunk == _firstChunk)
+                    return false;
+
+                if (_manyChunks != null)
+                    return _manyChunks.MoveNext(ref _currentChunk);
+
+                StringBuilder next = _firstChunk;
+                while (next.m_ChunkPrevious != _currentChunk)
+                    next = next.m_ChunkPrevious;
+                _currentChunk = next;
+                return true;
+            }
+            /// <summary>
+            /// Implements the IEnumerator pattern.
+            /// </summary>
+            public ReadOnlyMemory<char> Current => new ReadOnlyMemory<char>(_currentChunk.m_ChunkChars, 0, _currentChunk.m_ChunkLength);
+
+            #region private
+            internal ChunkEnumerator(StringBuilder stringBuilder)
+            {
+                Debug.Assert(stringBuilder != null);
+                _firstChunk = stringBuilder;
+                _currentChunk = null;   // MoveNext will find the last chunk if we do this.  
+                _manyChunks = null;
+
+                // There is a performance-vs-allocation tradeoff.   Because the chunks
+                // are a linked list with each chunk pointing to its PREDECESSOR, walking
+                // the list FORWARD is not efficient.   If there are few chunks (< 8) we
+                // simply scan from the start each time, and tolerate the N*N behavior. 
+                // However above this size, we allocate an array to hold pointers to all
+                // the chunks and we can be efficient for large N.    
+                int chunkCount = ChunkCount(stringBuilder);
+                if (8 < chunkCount)
```

Comment:
Nit, this is sort of a Yoda conditional

Reply:
I don't follow the change you want.

Reply:
`if (chunkCount > 8)` https://en.wikipedia.org/wiki/Yoda_conditions

Reply:
I see.   Quite a while back, I adopted a convention of always making relational operators go from smallest to largest.   (basically I never use > or >=).    I find it slightly clearer and less error prone, as it eliminates one source of arbitrary variation, and allows visual position to help in reading the code (this is especially nice if you are testing for a range).   

For what it is worth...

Reply:
That makes sense. I may do it.